test/e2e/play_kube_test.go

```diff
@@ -6310,4 +6310,85 @@ spec:
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(ExitWithError(125, "invalid signal: noSuchSignal"))
 	})
+
+	It("test with custom log path from containers.conf", func() {
+		customLogPath := filepath.Join(podmanTest.TempDir, "custom-logs")
+		expectedMessage := "Pod started, checking logs from test"
+
+		conffile := filepath.Join(podmanTest.TempDir, "containers.conf")
+		configContent := fmt.Sprintf(`[containers]
+log_driver = "k8s-file"
+log_path = "%s"
+`, customLogPath)
+
+		err := os.WriteFile(conffile, []byte(configContent), 0644)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = os.MkdirAll(customLogPath, 0755)
+		Expect(err).ToNot(HaveOccurred())
+
+		oldContainersConf := os.Getenv("CONTAINERS_CONF")
+		os.Setenv("CONTAINERS_CONF", conffile)
+		defer os.Setenv("CONTAINERS_CONF", oldContainersConf)
+
+		if IsRemote() {
+			podmanTest.RestartRemoteService()
+		}
+
+		kubeYaml := filepath.Join(podmanTest.TempDir, "test-pod.yaml")
+		podYamlContent := fmt.Sprintf(`apiVersion: v1
+kind: Pod
+metadata:
+  name: log-test-pod
+spec:
+  containers:
+  - name: logger-container
+    image: %s
+    command: ["/bin/sh", "-c", "echo '%s'; sleep 2"]
+`, CITEST_IMAGE, expectedMessage)
+
+		err = os.WriteFile(kubeYaml, []byte(podYamlContent), 0644)
+		Expect(err).ToNot(HaveOccurred())
+
+		session := podmanTest.Podman([]string{"kube", "play", kubeYaml})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
+
+		session = podmanTest.Podman([]string{"wait", "log-test-pod-logger-container"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
+
+		customLogDirs, err := os.ReadDir(customLogPath)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(customLogDirs).To(HaveLen(2), "Should have exactly two container log directories (infra + app container)")
+
+		var appContainerLogDir string
+		var logContent string
+
+		Eventually(func() bool {
+			for _, dir := range customLogDirs {
+				if !dir.IsDir() {
+					continue
+				}
+				containerLogDir := dir.Name()
+				logFilePath := filepath.Join(customLogPath, containerLogDir, "ctr.log")
+				if _, err := os.Stat(logFilePath); err != nil {
+					continue
+				}
+				content, err := os.ReadFile(logFilePath)
+				if err != nil {
+					continue
+				}
+				if strings.Contains(string(content), expectedMessage) {
+					appContainerLogDir = containerLogDir
+					logContent = string(content)
+					return true
+				}
+			}
+			return false
+		}, "10s", "100ms").Should(BeTrue(), "Should find log file with expected message within timeout")
```

Comment:
There is no need to wrap this into eventually, just match these things directly. Because you use wait this must pass right away.